modules/stereo/src/matching.hpp

```diff
@@ -0,0 +1,364 @@
+                                for (int l = j + 1; l < 9; ++l)
+                                    if (window[l] < window[min])
+                                        min = l;
+                                const uint8_t temp = window[j];
+                                window[j] = window[min];
+                                window[min] = temp;
+                            }
+                            mapModified[m  * width + n] = window[4];
+                        }
+                    }
+                }
+            };
+            //!median 9x1 paralelized filter
+            class Median9x1:public ParallelLoopBody
+            {
+            private:
+                uint8_t *harta;
+                uint8_t *mapModified;
+                int height, width;
+            public:
+                Median9x1(const Mat &hartaOriginala, Mat &map)
+                {
+                    harta = hartaOriginala.data;
+                    mapModified = map.data;
+                    height = hartaOriginala.rows;
+                    width = hartaOriginala.cols;
+                }
+                void operator()(const Range &r) const{
+                    for (int n = r.start; n <= r.end; ++n)
+                    {
+                        for (int m = 4; m < height - 4; ++m)
+                        {
+                            int k = 0;
+                            uint8_t window[9];
+                            for (int i = m - 4; i <= m + 4; ++i)
+                                window[k++] = harta[i * width + n];
+                            for (int j = 0; j < 5; j++)
+                            {
+                                int min = j;
+                                for (int l = j + 1; l < 9; ++l)
+                                    if (window[l] < window[min])
+                                        min = l;
+                                const uint8_t temp = window[j];
+                                window[j] = window[min];
+                                window[min] = temp;
+                            }
+                            mapModified[m  * width + n] = window[4];
+                        }
+                    }
+                }
+            };
+        public:
+            //!method for setting the maximum disparity
+            void setMaxDisparity(int val);
+            //!method for getting the disparity
+            int getMaxDisparity();
+            //!method for setting the scalling factor
+            void setScallingFactor(int val);
+            //!method for getting the scalling factor
+            int getScallingFactor();
+            //!setter for the confidence check
+            void setConfidence(double val);
+            //!getter for confidence check
+            double getConfidence();
+            //!method for computing the hamming difference
+            void hammingDistanceBlockMatching(const Mat &left, const Mat &right, Mat &c, const int kernelSize = 9);
```

Comment:
Are all of these methods supposed to be used by the client code? I don't think it's a good idea. 

Also, it seems like a simple "here's 2 images, give me the disparity" is missing, isn't it?